includes/REST_API/Stories_Media_Controller.php

```diff
@@ -69,6 +69,33 @@ public function get_items( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Updates a single attachment.
+	 *
+	 * Override the existing method so we can set parent id.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+
+		if ( $request['web_stories_parent'] && get_post( $request['web_stories_parent'] ) ) {
+			$args   = [
+				'ID'          => $request['id'],
+				'post_parent' => $request['web_stories_parent'],
+			];
+			$result = wp_update_post( $args, true );
+			if ( is_wp_error( $result ) ) {
+				return $result;
+			}
+		}
+
+		return parent::update_item( $request );
```

Comment:
Should this happen before the above logic? If there was an error in `parent::update_item( $request )` we should not proceed with our custom logic.